subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/extensions/AbstractMultiTestInterceptor.java

```diff
@@ -58,31 +60,82 @@ protected AbstractMultiTestInterceptor(Class<?> target, boolean runAllExecutions
     public void interceptFeature(FeatureInfo feature) {
         initExecutions();
 
+        boolean featureIsParameterized = feature.isParameterized();
+        if (!featureIsParameterized) {
+            parameterizeFeature(feature);
+        }
+
         NameProvider<IterationInfo> iterationNameProvider = feature.getIterationNameProvider();
         if (iterationNameProvider == null) {
-            feature.setName(feature.getName() + " " + executions);
+            feature.setIterationNameProvider(p -> feature.getName() + " " + currentExecution);
         } else {
-            feature.setIterationNameProvider(p -> iterationNameProvider.getName(p) + " " + executions);
+            feature.setIterationNameProvider(iteration -> iterationNameProvider.getName(iteration) + " " + currentExecution);
         }
 
+        feature.setDataDriver((dataIterator, iterationRunner, parameters) -> {
+            TestDetails testDetails = new TestDetails(feature);
+            while(dataIterator.hasNext()) {
+                Object[] arguments = dataIterator.next();
+                Object[] actualArguments =  featureIsParameterized ? IDataDriver.prepareArgumentArray(arguments, parameters) : new Object[0];
+                for (Execution execution : executions) {
+                    if (execution.isTestEnabled(testDetails)) {
+                        currentExecution = execution;
+                        try {
+                            iterationRunner.runIteration(actualArguments);
+                        } catch (Throwable t) {
+                            currentExecution = null;
+                        }
+
+                        if (!runAllExecutions) {
+                            break;
+                        }
+                    }
+                }
+            }
+        });
+
         if (canSkipFeature(feature, executions)) {
             feature.skip(executions.toString());
             return;
         }
 
-        if (feature.isParameterized()) {
-            feature.getIterationInterceptors().add(0, new ParameterizedFeatureMultiVersionInterceptor());
-        } else {
-            IMethodInterceptor interceptor = new NonParameterizedFeatureMultiVersionInterceptor();
-            feature.addInterceptor(interceptor);
-            feature.addIterationInterceptor(interceptor);
-        }
+        feature.getIterationInterceptors().add(0, new ParameterizedFeatureMultiVersionInterceptor());
     }
 
-    @Override
-    public void interceptInitializerMethod(IMethodInvocation invocation) throws Throwable {
-        this.initInvocation = invocation;
-        initInvocation.proceed();
+    private void parameterizeFeature(FeatureInfo feature) {
```

Comment:
IIUC, this makes a dummy method with 1 iteration and in that iteration there's zero parameter, right?

Reply:
Yes.